GenderPayGap.WebUI/BackgroundJobs/ScheduledJobs/AnonymiseRetiredDuplicateUsersJob.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Linq;
+using GenderPayGap.Core;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database;
+using GenderPayGap.Database.Models;
+using GenderPayGap.WebUI.Helpers;
+using GenderPayGap.WebUI.Services;
+
+namespace GenderPayGap.WebUI.BackgroundJobs.ScheduledJobs
+{
+    public class AnonymiseRetiredDuplicateUsersJob
+    {
+
+        private readonly AuditLogger auditLogger;
+        private readonly IDataRepository dataRepository;
+
+        public AnonymiseRetiredDuplicateUsersJob(IDataRepository dataRepository)
+        {
+            this.dataRepository = dataRepository;
+        }
+
+
+        //Remove any duplicate retired user details
+        public void AnonymiseRetiredDuplicateUsers()
+        {
+            JobHelpers.RunAndLogJob(AnonymiseRetiredDuplicateUsersAction, nameof(AnonymiseRetiredDuplicateUsers));
+        }
+
+        private void AnonymiseRetiredDuplicateUsersAction()
+        {
+            List<User> users = dataRepository.GetAll<User>().ToList();
+
+            var usersWithMultipleAccounts = users
+                .Where(u => u.HasBeenAnonymised == false)
+                .GroupBy(u => u.EmailAddress).Where(e => e.Count() > 1)
+                .SelectMany(g => g)
+                .ToList();
+
+            foreach (var user in usersWithMultipleAccounts.Where(u => u.Status == UserStatuses.Retired))
```

Comment:
we also anonymise the users that have more retired accounts (no active one). I'm wondering if we need to keep the data for those users

Reply:
Do you mean something like, if all retired anonymise all but the most recently created account?

Reply:
yes, just in case that data is needed for something (I'm not sure if it is). What do you think?

Reply:
Yeah, I wasn't 100% sure tbh, I did change my mind a few times. I feel like the data is not needed really, but this does feel more complete and considered, removing only the minimum amount of data